src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2362,7 +2325,7 @@ because xbuild doesn't support framework reference assemblies.
   <GeneratePackageManagerJava
     ResolvedAssemblies="@(_ResolvedAssemblies)"
     ResolvedUserAssemblies="@(_ResolvedUserAssemblies)"
-    MainAssembly="$(MonoAndroidLinkerInputDir)$(TargetFileName)"
+    MainAssembly="$(TargetPath)"
```

Comment:
should this not be using `$(_JniMarshalMethodsInputDir)$(TargetFileName)` if `$(AndroidGenerateJniMarshalMethods) == 'true'` ?

Reply:
This was looking at the assembly in `linksrc` before, which was basically the file copied from `$(TargetPath)`. Does this actually need to look at the assembly *after* it was linked?